install_viewer.py

```diff
@@ -0,0 +1,113 @@
+import os
+import json
+import zipfile
+import shutil
+import sys
+
+from urllib.request import urlretrieve, urlopen
+
+#def downloadFile(url, outFile):
+#    urlretrieve(url, outFile)
+
+def isNewer(new, original):
+    majorMult, minorMult, patchMult = 10000,100,1
+    major1, minor1, patch1 = original.split('.')
+    major2, minor2, patch2 = new.split('.')
+    return majorMult * major2 + minorMult * minor2 + patchMult * minor2 > majorMult * major1 + minorMult * minor1 + patchMult * minor1
+
+
+def main():
+    currentInstallFileName = "viewer_currentInstall.json"
+
+    versionAndChangeLogUrl = "https://s3.amazonaws.com/battlecode-2018/viewer/"
+
+    versionFileName = "version.txt"
+    changelogFileName = "changelog.json"
+
+    baseUrl = "https://s3.amazonaws.com/battlecode-2018/viewer/"
+
+    directory = os.path.dirname(os.path.realpath(__file__))
+
+    zipFileName = "viewer_latest.zip"
+    viewerDirectory = "viewer_latest/"
+
+    currentInfoFileLocation = os.path.join(directory, currentInstallFileName)
+    if os.path.exists(currentInfoFileLocation):
+        currentInfoFile = open(currentInfoFileLocation)
+        currentInfo = json.load(currentInfoFile)
+        currentInfoFile.close()
+    else:
+        possibleSystems = [
```

Comment:
Use [platform.architecture()](https://docs.python.org/3/library/platform.html#platform.architecture) and and [platform.system()](https://docs.python.org/3/library/platform.html#platform.system)

Reply:
These function have inconsistent returns, and as the user is only ever prompted once, I don't think this is a problem.